### PR TITLE
Order imports when reformatting

### PR DESCRIPTION
**Pull request by pugnascotia**
Change the formatter config to sort / order imports, and reformat the codebase. We already had a config file for Eclipse users, so Spotless now uses that.

The "Eclipse Code Formatter" plugin ought to be able to use this file as well for import ordering, but in my experiments the results were poor. Instead, use IntelliJ's `.editorconfig` support to configure import ordering.

I've also added a config file for the formatter plugin.

Other changes:
   * I've quietly enabled the `toggleOnOff` option for Spotless. It was already possible to disable formatting for sections using the markers for docs snippets, so enabling this option just accepts this reality and makes it possible via `formatter:off` and `formatter:on` without the restrictions around line length. It should still only be used as a very last resort and with good reason.
   * I've removed mention of the `paddedCell` option from the contributing guide, since I haven't had to use that option for a very long time. I moved the docs to the spotless config.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -194,11 +198,37 @@ Code Formatter] installed, you can apply formatting directly in IntelliJ.
...
+
+It is also important that you configure the correct import ordering in
+IntelliJ.  We configure Spotless (above) to sort the imports in Java files,
+and while the [Eclipse Code Formatter] plugin supports an option for
+ordering imports, in practice this option appears unreliable, so instead
+you must configure IntelliJ's import ordering to match Spotless.
+
+   1. Open **Preferences > Editor > Code Style > Java**
+   2. Click the **Imports** tab
+   3. Set "Class count to use import with '*'" to 9999.
```
Comment (mark-vieira) [inline CONTRIBUTING.md:214]: The class count bit should be done by the config in `elasticsearch.ide.gradle` on import.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -194,11 +198,37 @@ Code Formatter] installed, you can apply formatting directly in IntelliJ.
 
 Note that only some sub-projects in the Elasticsearch project are currently
 fully-formatted. You can see a list of project that **are not**
-automatically formatted in [gradle/formatting.gradle](gradle/formatting.gradle).
+automatically formatted in
+[build-tools-internal/src/main/groovy/elasticsearch.formatting.gradle](build-tools-internal/src/main/groovy/elasticsearch.formatting.gradle).
+
+#### Imports
+
+It is also important that you configure the correct import ordering in
```
Comment (mark-vieira) [inline CONTRIBUTING.md:206]: Is there really no way to automate setting this up somehow? A lot of the other manual stuff is sort of "optional" but optimizing imports is something _everyone_ does and if someone misses this step they are going to be fighting with spotless _a  lot_.

**Review by mark-vieira — COMMENTED** [review]
FWIW I'm a bit on the fence on this one unless we can have this configured on import automatically. A lot of this manual IDE config stuff is missed by developers. If someone doesn't do this they are going to be in for a very frustrating time battling between IDE rules and spotless.

I'll quote David here:
> yes sorting differently from IntelliJ would definitely be worse than today’s lack of sorting

We need to factor in that if this isn't configured automatically a non-negligible number of folks are going to hit this issue. I think having manual setup for things that are "optional" is fine, but I consider having import ordering rules match between the IDE and build be pretty much mandatory given that basically everyone uses this IDE feature.

**Comment by mark-vieira** [comment]
I think we can commit some stuff to `.idea/codeStyles` to declare a code style config with this stuff and have intelliJ automatically pick it up on import. We can then just ditch the `codeStyle { }` block in `elasticsearch.ide.gradle`.

**Comment by mark-vieira** [comment]
So yeah, let's set up the IDE import config the way we want, export it to a file in `.idea/codeStyles` and then edit `.idea/codestyles/codeStyleConfig.xml` to point to that config. I think that should do the trick.

**Comment by pugnascotia** [comment]
@mark-vieira I tried your suggestion but struggled to get it to work. I asked in the [IntelliJ forums](https://intellij-support.jetbrains.com/hc/en-us/community/posts/360010865000-Storing-code-style-in-project-in-Ultimate-2021-1) and someone suggested using a Settings Repository. Not mad keen on introducing something new - what do you think?

**Comment by mark-vieira** [comment]
Yeah, that is not gonna be a good solution given that code style is a _project_ setting and settings repositories are for making _all_ your IDE settings portable. I mean, we commit other stuff in the .idea folder, I feel like this _should_ work. I wonder if we need to do something like we do for the Gradle JVM settings and _modify_ the `codeStyleConfig.xmll` config file on import vs check one into source control. I assume that's the issue? That the IDE is _fighting_ with what we set there? I presume that just dropping a custom codestyle file in there get's picked up by IntelliJ just fine.

**Comment by pugnascotia** [comment]
So it turns out you can configure the imports order in IntelliJ via `.editorconfig`, so that's what I've done for Java and Groovy.

To see all supported properties, right-click a folder and select "New | EditorConfig File", check the IntelliJ IDEA-specific checkbox in the dialog that opens and, if needed, select the languages used in your project. An `.editorconfig` file will be generated with all IntelliJ-specific properties listed as comments.

**Review by mark-vieira — APPROVED** [review]
Brilliant :+1:

**Comment by pugnascotia** [comment]
Backported to 7.x in fb8f84fdaeb

**Pull request merged**